portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,24 +35,37 @@
     
     public class Name 
     {
-        private ArrayList<String> names = new ArrayList<String>();
+        public ArrayList<String> names = new ArrayList<String>();
     }
   
-    private Name name = new Name();
+    Name name = new Name();
+   
 
-  
+    
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Name name = new Name();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Gson gson = new Gson(); 
+        Query query = new Query("Names");
+        PreparedQuery results = datastore.prepare(query);
+        
+        for (Entity entity : results.asIterable()) 
+        {
+            name.names.add( (String) entity.getProperty("viewer0"));
+        }
```

Comment:
the Get method reads in all the names from the server and saves them in the name.names array. Every time the page is refreshed the names must be read in again. This seems redundant and I'm not sure how to make this more efficient.  The reason I use an instance of the name class is because it can be easily converted into JSON and returned by the get method.